source/jormungandr/jormungandr/scenarios/ridesharing/ridesharing_service_manager.py

```diff
@@ -0,0 +1,342 @@
+
+    def _init_class(self, cls, arguments):
+        """
+        Create an instance of a provider according to config
+        :param cls: provider class in Jormungandr found in config file
+        :param arguments: parameters to set in the provider class
+        :return: instance of provider
+        """
+        try:
+            if '.' not in cls:
+                self.logger.warning('impossible to build, wrongly formated class: {}'.format(cls))
+
+            module_path, name = cls.rsplit('.', 1)
+            module = import_module(module_path)
+            attr = getattr(module, name)
+            return attr(**arguments)
+        except ImportError:
+            self.logger.warning('impossible to build, cannot find class: {}'.format(cls))
+
+    def _update_ridesharing_service(self, service):
+        self.logger.info('updating/adding {} ridesharing service'.format(service.id))
+        try:
+            self._ridesharing_services[service.id] = self._init_class(service.klass, service.args)
+            self._ridesharing_services_last_update[service.id] = service.last_update()
+        except Exception:
+            self.logger.exception('impossible to initialize ridesharing service')
+
+        # If the provider added in db is also defined in legacy, delete it.
+        services_from_db = list(self._ridesharing_services.values())
+        services_legacy = self._ridesharing_services_legacy
+        if services_legacy and services_from_db:
+            new_services_legacy = []
+            for srv in services_legacy:
+                if srv not in services_from_db:
+                    new_services_legacy.append(srv)
+            self._ridesharing_services_legacy = new_services_legacy
+
+    def update_config(self):
+        """
+        Update list of ridesharing services from db
+        """
+        if (
+            self._last_update + datetime.timedelta(seconds=self._update_interval) > datetime.datetime.utcnow()
+            or not self._rs_services_getter
+        ):
+            return
+
+        self.logger.debug('Updating ridesharing services from db')
+        self._last_update = datetime.datetime.utcnow()
+
+        services = []
+        try:
+            services = self._rs_services_getter()
+        except Exception:
+            self.logger.exception('Failure to retrieve ridesharing services configuration')
+        if not services:
+            self.logger.debug('No ridesharing services disabled in db')
+            self._ridesharing_services = {}
+            self._ridesharing_services_last_update = {}
+
+            return
+
+        for service in services:
+            if (
+                service.id not in self._ridesharing_services_last_update
+                or service.last_update() > self._ridesharing_services_last_update[service.id]
+            ):
+                self._update_ridesharing_service(service)
+
+    def ridesharing_services_activate(self):
```

Comment:
Maybe
```suggestion
    def ridesharing_services_activated(self):
```
as the function returns `True` if some service is activated, and `False` otherwise ?

Reply:
:+1: